engine/api/get-started.md

```diff
@@ -606,6 +608,101 @@ $ curl --unix-socket /var/run/docker.sock \
 </div>
 </div> <!-- end tab-content -->
 
+
+### Pull images with authentication
+
+Pull images, like `docker pull`, with authentication:
+
+> **Note**: Credentials are sent in the clear. Docker's official registries use
+> HTTPS. Private registries should also be configured to use HTTPS.
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" data-target="#tab-pullimages-auth-python" data-group="python">Python</a></li>
+  <li><a data-toggle="tab" data-target="#tab-pullimages-auth-go" data-group="go">Go</a></li>
+  <li><a data-toggle="tab" data-target="#tab-pullimages-auth-curl" data-group="curl">Curl</a></li>
+</ul>
+
+<div class="tab-content">
+<div id="tab-pullimages-auth-python" class="tab-pane fade in active" markdown="1">
+
+```python
+import docker
+client = docker.from_env()
+image = client.images.pull("alpine", auth_config=['username','password'])
+print image.id
+```
```

Comment:
Hmm, the Python SDK retrieves auth info from the `~/.docker/config.json` file (and has integration with [credential helpers](https://github.com/docker/docker-credential-helpers)).

If for some reason you want to use a different set of credentials for this session, the easiest way to do so is as follows:
```python
import docker

client = docker.from_env()
client.login('username', 'password')
image = client.images.pull('alpine')
print(image.id)
```
The credentials provided in the call to `client.login` will be used for the rest of the session.

If one wants to use a specific credentials for a single pull, then `auth_config` parameter should be used as follows:

```python
import docker

client = docker.from_env()
image = client.images.pull('alpine', auth_config={"Username": "username", "Password": "password"})
print(image.id)
```

But honestly that second example is more advanced use (it bypasses a lot of checks and functionalities of the library) and probably doesn't belong on an introductory page.